contrib/submit-queue/github/github_test.go

```diff
@@ -0,0 +1,289 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-github/github"
+)
+
+func stringPtr(val string) *string { return &val }
+
+func TestHasLabel(t *testing.T) {
+	tests := []struct {
+		labels   []github.Label
+		label    string
+		hasLabel bool
+	}{
+		{
+			labels: []github.Label{
+				{Name: stringPtr("foo")},
+			},
+			label:    "foo",
+			hasLabel: true,
+		},
+		{
+			labels: []github.Label{
+				{Name: stringPtr("bar")},
+			},
+			label:    "foo",
+			hasLabel: false,
+		},
+		{
+			labels: []github.Label{
+				{Name: stringPtr("bar")},
+				{Name: stringPtr("foo")},
+			},
+			label:    "foo",
+			hasLabel: true,
+		},
+		{
+			labels: []github.Label{
+				{Name: stringPtr("bar")},
+				{Name: stringPtr("baz")},
+			},
+			label:    "foo",
+			hasLabel: false,
+		},
+	}
+
+	for _, test := range tests {
+		if test.hasLabel != hasLabel(test.labels, test.label) {
+			t.Errorf("Unexpected output: %v", test)
+		}
+	}
+}
+
+func TestHasLabels(t *testing.T) {
+	tests := []struct {
+		labels     []github.Label
+		seekLabels []string
+		hasLabel   bool
+	}{
+		{
+			labels: []github.Label{
+				{Name: stringPtr("foo")},
+			},
+			seekLabels: []string{"foo"},
+			hasLabel:   true,
+		},
+		{
+			labels: []github.Label{
+				{Name: stringPtr("bar")},
+			},
+			seekLabels: []string{"foo"},
+			hasLabel:   false,
+		},
+		{
+			labels: []github.Label{
+				{Name: stringPtr("bar")},
+				{Name: stringPtr("foo")},
+			},
+			seekLabels: []string{"foo"},
+			hasLabel:   true,
+		},
+		{
+			labels: []github.Label{
+				{Name: stringPtr("bar")},
+				{Name: stringPtr("baz")},
+			},
+			seekLabels: []string{"foo"},
+			hasLabel:   false,
+		},
+		{
+			labels: []github.Label{
+				{Name: stringPtr("foo")},
+			},
+			seekLabels: []string{"foo", "bar"},
+			hasLabel:   false,
+		},
+	}
+
+	for _, test := range tests {
+		if test.hasLabel != hasLabels(test.labels, test.seekLabels) {
+			t.Errorf("Unexpected output: %v", test)
+		}
+	}
+}
+
+func initTest() (*github.Client, *httptest.Server, *http.ServeMux) {
+	// test server
+	mux := http.NewServeMux()
+	server := httptest.NewServer(mux)
+
+	// github client configured to use test server
+	client := github.NewClient(nil)
+	url, _ := url.Parse(server.URL)
+	client.BaseURL = url
+	client.UploadURL = url
+
+	return client, server, mux
+}
+
+func TestFetchAllPRs(t *testing.T) {
+	tests := []struct {
+		PullRequests [][]github.PullRequest
+		Pages        []int
+	}{
+		{
+			PullRequests: [][]github.PullRequest{
+				{
+					{},
+				},
+			},
+			Pages: []int{0},
+		},
+		{
+			PullRequests: [][]github.PullRequest{
+				{
+					{},
+				},
+				{
+					{},
+				},
+				{
+					{},
+				},
+				{
+					{},
+				},
+			},
+			Pages: []int{4, 4, 4, 0},
+		},
+		{
+			PullRequests: [][]github.PullRequest{
```

Comment:
You're on thin ice, pal.

On Mon, Jul 27, 2015 at 9:33 PM, Paul Morie <notifications@github.com>
wrote:

> In contrib/submit-queue/github/github_test.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11844#discussion_r35613407>
> :
>
> > +					{},
> > +				},
> > +				{
> > +					{},
> > +				},
> > +				{
> > +					{},
> > +				},
> > +				{
> > +					{},
> > +				},
> > +			},
> > +			Pages: []int{4, 4, 4, 0},
> > +		},
> > +		{
> > +			PullRequests: [][]github.PullRequest{
>
> Brace yourself
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11844/files#r35613407>
> .
>